tests/http_basicauth.test.js

```diff
@@ -0,0 +1,15 @@
+import child_process from "child_process";
+import fs from "fs";
+
+test("test that http basic auth works", async () => {
+  child_process.execSync(
+    'docker run -v $PWD/test-crawls:/crawls webrecorder/browsertrix-crawler crawl --url http://httpbin.org/basic-auth/foo/bar --httpBasicAuth foo:bar --collection basicauth-test --statsFilename stats.json',
```

Comment:
httpbin.org doesn't have a responding maintainer at the moment. I think it's better for test stability to boot https://github.com/psf/httpbin with docker instead of relying on external hosting.

Reply:
Agreed but might be made complicated by https://github.com/psf/httpbin/issues/27

Reply:
Yeah, I noticed the Docker image wasn't currently working.

I was going to suggest https://github.com/mccutchen/go-httpbin (I'm not sure if it does http auth). But I wasn't entirely sure how to slot in running the container as part of the current test suite.

Reply:
Couldn't find support for http auth in go-httpbin. Another option would be to boot a webserver inside the test suite like here: https://github.com/webrecorder/browsertrix-crawler/blob/main/src/util/replayserver.ts#L28-L64

Reply:
I pushed up a revised test that starts a web server and then ensures browsertrix-crawler can access it. Unfortunately it doesn't seem to work as written (even when I'm not yet checking username/password). 

The crawler gets reports:

```
{"timestamp":"2024-05-16T17:50:10.725Z","logLevel":"error","context":"general","message":"Page Load Timeout, skipping page","details":{"msg":"net::ERR_CONNECTION_REFUSED at http://host.docker.internal:9998/","page":"http://host.docker.internal:9998/","workerid":0}}
```

But I can see when running the server and the docker container independently (independent of the test) that things work ok with `http://host.docker.internal:9998`. So I'm a bit stumped what I'm doing wrong here.

Reply:
@edsu I think beforeAll is expecting a promise to be returned, so might be a timing/async issue as you suspect. Might be easier to just start the server in-line prior to running the crawler and close after it's done, all within the test?

Reply:
FWIW, I believe it has to be the docker IP 172.17.0.1 while host.docker.internal is more for mac/windows.